### ISSUE TITLE
[CLI]: `WandbCallback` was not able to get prediction samples -> Object of type MultiCategory is not JSON serializable

### ISSUE THREAD

**Issue by deven-gqc** [wandb/wandb#4290]
### Describe the bug

I have a simple training loop in `fastai` which does multi-label image classification. I have a `wandb` callback attached to the `Learner` object. 

![image](https://user-images.githubusercontent.com/86305049/191335261-efeda3a6-705d-48a1-b2c0-0a390050524e.png)

However, the callback does not log any predictions from the object, it says that:
```
WandbCallback was not able to get prediction samples -> Object of type MultiCategory is not JSON serializable
```

### Environment

WandB version: 0.13.3 

OS: Ubuntu

Python version: 3.7.14

Versions of relevant libraries: 
`fastai`: 2.7.9

**Comment by thanos-wandb** [comment]
Hi @deven-gqc thank you for reporting this! Could you please try the following and let me know if this would work for you?

`learn = vision_learner(..., cbs=WandbCallback())`

**Comment by deven-gqc** [comment]
Hey @thanos-wandb, I tried that, however, the issue still remains the same.

![image](https://user-images.githubusercontent.com/86305049/191642523-61ccf237-a42c-4fb3-b288-9f0c449faf6e.png)

**Comment by deven-gqc** [comment]
@thanos-wandb Any update on this?

**Comment by thanos-wandb** [comment]
Hi @deven-gqc sorry for the late response to this one. The error seems to occur because of the metrics type not being supported. Can you please try running your code again for an epoch without the `metrics` and see if that would execute without any errors? also it would help to provide us with a code snippet on how you defined the `f1_*` metrics.

**Comment by deven-gqc** [comment]
@thanos-wandb The metrics are defined as follows, `F1ScoreMulti` is imported from `scikit-learn`
```
f1_macro = F1ScoreMulti(thresh=0.5, average='macro')
f1_macro.name = 'F1(macro)'
f1_samples = F1ScoreMulti(thresh=0.5, average='samples')
f1_samples.name = 'F1(samples)'
```

The issue still remains.
![image](https://user-images.githubusercontent.com/86305049/194583609-166e5904-1f1c-4f55-8b1b-29f9bba45e2d.png)

I think the issue stems from the `TensorMultiCategory` object in `fastai`, the problem does not arise from the metrics. A `TensorMultiCategory` object gets created for a multi-label classification problem.
![image](https://user-images.githubusercontent.com/86305049/194584361-61110570-2f3c-4b8c-8626-933126a2096a.png)

**Comment by thanos-wandb** [comment]
Hi @deven-gqc sorry for the late response to this one, could you please add the following argument to prevent from logging predictions in a Table:
` learn = vision_learner(..., cbs=WandbCallback(log_preds=False))`
Would this work for you?

**Comment by deven-gqc** [comment]
@thanos-wandb I did that and the error goes away.
![image](https://user-images.githubusercontent.com/86305049/195339129-f2972f9f-046f-4378-8f3f-48ce85c95e97.png)

**Comment by thanos-wandb** [comment]
Hi @deven-gqc I am glad this worked for you and thanks for the confirmation. As a final test, would it be possible to check now with the same `metrics` you had before?
`learn = vision_learner(dls, resnet50, metrics=..., cbs=WandbCallback(log_preds=False))`

**Comment by deven-gqc** [comment]
@thanos-wandb It works. 

![image](https://user-images.githubusercontent.com/86305049/195598206-9cef0066-5beb-4524-9d9d-d5cc1e1a35db.png)

**Comment by thanos-wandb** [comment]
Hi @deven-gqc happy to hear that at least the training process now works. 

Regarding the logging, can you please check the format of your prediction samples what type and shape of data are these? Please also have a look at [this report](https://wandb.ai/borisd13/demo_config/reports/Visualize-Track-and-Compare-fastai-Models-With-Weights-Biases--Vmlldzo4MzAyNA) and the accompanying Colab in case it helps.

**Comment by deven-gqc** [comment]
@thanos-wandb Training worked before as well. The only thing that didn't work was the logging of prediction images from the model. 

When I use the predict method, I get 3 things back. 
1. names of the classes
2. whether those objects are present in the image or not (type (TensorBase))
3. probabilities for all those classes (type (TensorBase))
![image](https://user-images.githubusercontent.com/86305049/196225366-b27b9432-c25f-475d-a5e4-eea24554160b.png)

**Comment by thanos-wandb** [comment]
Hi @deven-gqc thanks for this additional context, I will need to reproduce these types on my end and get back to you. We can make a feature request for this integration, if there is no other workaround.

**Comment by deven-gqc** [comment]
@thanos-wandb sounds good to me, please let me know if you need any additional input.

**Comment by thanos-wandb** [comment]
Hi @deven-gqc sorry for the late reply, just an update about this issue I filed a feature request to the engineering team to support `TensorMultiCategory` prediction samples to be logged using WandbLogger. We will reach out to you here when there are any updates about it. In the meantime, if it's possible to convert the prediction samples into images and string types you could log them using `wandb.Table`. I hope this helps.
